interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java

```diff
@@ -203,18 +214,22 @@ public void setNotServing(
     }
   }
 
-  private static class HostnameInterceptor implements ServerInterceptor {
+  private static class TestInfoInterceptor implements ServerInterceptor {
     private final String host;
+    private final AtomicBoolean keepOpenCaptor;
 
-    private HostnameInterceptor(String host) {
+    private TestInfoInterceptor(String host, AtomicBoolean keepOpenCaptor) {
       this.host = host;
+      this.keepOpenCaptor = keepOpenCaptor;
     }
 
     @Override
     public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
         ServerCall<ReqT, RespT> call,
         final Metadata requestHeaders,
         ServerCallHandler<ReqT, RespT> next) {
+      String callBehavior = requestHeaders.get(CALL_BEHAVIOR_KEY);
+      keepOpenCaptor.set("keep-open".equals(callBehavior));
```

Comment:
I don't see how the keepOpen atomic's value is correlated with this call after this interceptor sets it to true.

My reading of the spec for these tests was that, for an incoming call with `keep-open` in the metadata, _that_ call should be kept open by the server. As I understand the code here, instead _any_ call with `keep-open` in the metadata essentially flips a global boolean and all calls from that point on will be kept open (at least until another RPC comes without `keep-open`). Is that intended?

To set a value that would instead be scoped to this particular call, I think you could set a value in the context, as in https://github.com/GoogleCloudPlatform/traffic-director-grpc-examples/blob/master/java/src/main/java/io/grpc/examples/wallet/WalletInterceptors.java#L106 (there might be a better way to accomplish this)

Reply:
Oops, you are right. This is screwed up, I was too focused on how the test case will be running, in which all calls will be open after some point of time.

Reply:
Should be fixed now.